packages/x-date-pickers/src/internals/components/PickersPopper.tsx

```diff
@@ -360,6 +360,16 @@ export function PickersPopper(inProps: PickerPopperProps) {
   }, [onDismiss, open]);
 
   const lastFocusedElementRef = React.useRef<Element | null>(null);
+
+  const handleClickAway: OnClickAway = useEventCallback(() => {
+    lastFocusedElementRef.current = null;
```

Comment:
Fix the root cause why pickers where returning the focus back to the open button after being clicked away:

https://github.com/mui/mui-x/assets/4941090/44da61f3-78b9-4114-a041-ad6868e792a9